subsys/net/l2/openthread/openthread.c

```diff
@@ -424,32 +430,35 @@ static int openthread_init(struct net_if *iface)
 
 	__ASSERT(ot_context->instance, "OT instance is NULL");
 
+//	if (!IS_ENABLED(CONFIG_OPENTHREAD_RCP)) { // TODO: check whether required
 	if (IS_ENABLED(CONFIG_OPENTHREAD_SHELL)) {
 		platformShellInit(ot_context->instance);
 	}
 
-	if (IS_ENABLED(CONFIG_OPENTHREAD_NCP)) {
+	if (IS_ENABLED(CONFIG_OPENTHREAD_RCP)) {
 		otNcpInit(ot_context->instance);
 	}
 
-	if (!IS_ENABLED(CONFIG_OPENTHREAD_RAW)) {
-		otIp6SetEnabled(ot_context->instance, true);
-	}
+	if (!IS_ENABLED(CONFIG_OPENTHREAD_RCP)) {
+		if (!IS_ENABLED(CONFIG_OPENTHREAD_RAW)) {
+			otIp6SetEnabled(ot_context->instance, true);
+		}
 
-	if (!IS_ENABLED(CONFIG_OPENTHREAD_NCP)) {
```

Comment:
Same here, I'd rather make it just `!IS_ENABLED(CONFIG_OPENTHREAD_NCP) && !IS_ENABLED(CONFIG_OPENTHREAD_RCP)`, w/o further changes.